src/rules/base.jl

```diff
@@ -62,7 +62,37 @@
 @scalar_rule(cotd(x), -(π / oftype(x, 180)) * (1 + Ω^2))
 @scalar_rule(sech(x), -tanh(x) * Ω)
 @scalar_rule(csch(x), -coth(x) * Ω)
-@scalar_rule(hypot(x, y), (y / Ω, x / Ω))
+
+function frule(::typeof(hypot), x::Real...)
+    Ω = hypot(x...)
+    return Ω, Rule((Δ...) -> sum(Δ .* x) * inv(Ω))
+end
+
+function rrule(::typeof(hypot), x::Real...)
+    Ω = hypot(x...)
+    return Ω, map(x) do x_i
+        Rule(ΔΩ -> ΔΩ * (x_i / Ω))
+    end
+end
+
+function frule(::typeof(hypot), x...)
+    Ω = hypot(x...)
+    return Ω, WirtingerRule(
+        Rule((Δ...) -> sum(Δ .* conj.(x) * inv(2Ω))),
```

Comment:
This should be moved flush with the opening parenthesis with the second line aligned below it.
```julia
    return Ω, WirtingerRule(Rule((Δ...) -> sum(Δ .* conj.(x) * inv(2Ω))),
                            Rule((Δ...) -> sum(Δ .* x) * inv(2Ω)))
```
Again in this case, why multiply by `inv` instead of just dividing?

Reply:
See my comment above. It might make sense to overload `/` for `Zero()` and `One()` though.

Reply:
Yeah I think that would probably make sense.

Reply:
Would it be ok if I left it for now? I'm not too experienced with Cassette and there would be some edge cases to think about, so I think this deserves its own PR.

Reply:
I had a go at it. Could maybe be done more elegantly, but works for now.